default.go

```diff
@@ -49,7 +49,7 @@ const (
 	//  <subdomain> ::= <label> | <subdomain> "." <label>
 	//  var subdomain = /^[a-zA-Z](([-0-9a-zA-Z]+)?[0-9a-zA-Z])?(\.[a-zA-Z](([-0-9a-zA-Z]+)?[0-9a-zA-Z])?)*$/;
 	//  <domain> ::= <subdomain> | " "
-	HostnamePattern = `^[a-zA-Z](([-0-9a-zA-Z]+)?[0-9a-zA-Z])?(\.[a-zA-Z](([-0-9a-zA-Z]+)?[0-9a-zA-Z])?)*$`
+	HostnamePattern = `^[a-zA-Z0-9\p{S}](([a-zA-Z0-9-\p{S}]{0,63})(\.)){0,6}([a-zA-Z]){2,3}$`
```

Comment:
Need some justification on why the last part of the pattern is:
(i) constrained in length
(ii) cannot itself contain unicode characters

Examples that should work and don't with your regexp:
* www.example.info
* www.example.arpa
* www.example.onion
* www.example.ôlà

Reply:
I've updated the regex, thanks for the tip I wasn't aware that we even had non-ascii character TLD's so having forewarning means we don't have to figure this out again when a customer complains. I've also added 2 or more characters to the regex for the tld matching. again thankyou.